frame/uniswap-v2/src/lib.rs

```diff
@@ -351,6 +390,108 @@ pub mod pallet {
 		}
 	}
 
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(_n: T::BlockNumber) -> Weight {
+			let current_timestamp = T::Time::now();
+			Pools::<T>::iter()
+				.filter(|(pool_id, pool)| T::EnabledPoolTWAP::contains(&(pool.pair, *pool_id)))
+				.fold(Weight::zero(), |weight, (pool_id, _)| {
+					let rate_base = Self::do_get_exchange_rate(pool_id, false);
+					let rate_quote = Self::do_get_exchange_rate(pool_id, true);
+					match (rate_base, rate_quote) {
+						(Ok(rate_base), Ok(rate_quote)) => PriceCumulativesState::<T>::try_mutate(
+							pool_id,
+							|price_cumulatives| -> Result<Weight, DispatchError> {
+								match *price_cumulatives {
+									// Initialization
+									None => {
+										let (price_cumulative_base, price_cumulative_quote) = (
```

Comment:
as in uniswap

Reply:
okay, this seems bit long work, perhaps with pool creation we also need to handle adding entry to EnabledPoolTWAP if pool wants TWAP feature. I prefer it to be done in saperate PR as this will be implemented on pablo, and there are some changes planned for dex-router and pablo integration. 
So perhaps it is better we complete https://github.com/ComposableFi/composable/pull/841 as it adds basic structure for TWAP and then improve as you suggest.

Reply:
in uniswap code is very simple, and we can do too. right now it is complicated more than needed. see https://github.com/Uniswap/v2-periphery/blob/master/contracts/examples/ExampleOracleSimple.sol

Reply:
is there separate task to do TWAP? I am ok with that than

Reply:
once pablo is added to runtime dali then we are planning to remove all other dexs. So we need to improve on pablo only.